drivers/dac/ad5754r/ad5754r.h

```diff
@@ -0,0 +1,387 @@
+	AD5754R_DAC_CH_D,
+};
+
+/**
+ * @enum ad5754r_dac_ch_range
+ * @brief DAC Channel Output Ranges
+ */
+enum ad5754r_dac_ch_range {
+	AD5754R_SPAN_0V_TO_5V,
+	AD5754R_SPAN_0V_TO_10V,
+	AD5754R_SPAN_0V_TO_10V8,
+	AD5754R_SPAN_M5V_TO_5V,
+	AD5754R_SPAN_M10V_TO_10V,
+	AD5754R_SPAN_M10V8_TO_10V8,
+};
+
+/**
+ * @enum ad5754r_dac_addr_for_ctrl_settings
+ * @brief DAC Addresses for Control Register Settings
+ */
+enum ad5754r_dac_addr_for_ctrl_settings {
+	AD5754R_NOP_DAC_ADDR = 0,
+	AD5754R_TSD_CLAMP_CLR_SDO_DAC_ADDR = 1,
+	AD5754R_CLEAR_DAC_ADDR = 4,
+	AD5754R_LOAD_DAC_ADDR = 5
+};
+
+/**
+ * @enum ad5754r_ctrl_current_clamp_en
+ * @brief Current Clamp EN/DIS Modes
+ */
+enum ad5754r_ctrl_current_clamp_en {
+	AD5754R_CTRL_CLAMP_DIS,
+	AD5754R_CTRL_CLAMP_EN
+};
+
+/**
+ * @enum ad5754r_ctrl_clear_sel
+ * @brief Clear Mode Options
+ */
+enum ad5754r_ctrl_clear_sel {
+	AD5754R_CTRL_CLEAR_0V,
+	AD5754R_CTRL_CLEAR_MIDSCALE_CODE
+};
+
+/**
+ * @enum ad5754r_ctrl_sdo_dis
+ * @brief SDO EN/DIS Modes
+ */
+enum ad5754r_ctrl_sdo_dis {
+	AD5754R_CTRL_SDO_EN,
+	AD5754R_CTRL_SDO_DIS
+};
+
+/**
+ * @enum ad5754r_pwr_dac_ch_pwrup
+ * @brief DAC Channel Powerup Modes
+ */
+enum ad5754r_pwr_dac_ch_pwrup {
+	AD5754R_PWR_DAC_CH_POWERDOWN,
+	AD5754R_PWR_DAC_CH_POWERUP
+};
+
+/**
+ * @enum ad5754r_pwr_int_ref_pwrup
+ * @brief Internal Reference Powerup Modes
+ */
+enum ad5754r_pwr_int_ref_pwrup {
+	AD5754R_PWR_INT_REF_POWERDOWN,
+	AD5754R_PWR_INT_REF_POWERUP
+};
+
+/**
+ * @enum ad5754r_pwr_tsd_alert
+ * @brief Thermal Shutdown Alert State
+ */
+enum ad5754r_pwr_tsd_alert {
+	AD5754R_PWR_TSD_NOT_DETECTED,
+	AD5754R_PWR_TSD_DETECTED
+};
+
+/**
+ * @enum ad5754r_pwr_oc_ch_alert
+ * @brief DAC Channel Overcurrent Alert State
+ */
+enum ad5754r_pwr_oc_ch_alert {
+	AD5754R_PWR_OC_CH_NOT_DETECTED,
+	AD5754R_PWR_OC_CH_DETECTED
+};
+
+/**
+ * @enum ad5754r_encoding_scheme
+ * @brief DAC encoding scheme: Binary/2sComplement
+ */
+enum ad5754r_encoding_scheme {
+	AD5754R_ENCODING_TWOSCOMPLEMENT,
+	AD5754R_ENCODING_BINARY
+};
+
+/**
+ * @struct ad5754r_init_param
+ * @brief ad5754r Device initialization parameters.
+ */
+struct ad5754r_init_param {
+	/* SPI Init Parameters */
+	struct no_os_spi_init_param	*spi_init;
+	/* Clear GPIO Init Params*/
+	struct no_os_gpio_init_param *gpio_clear_init;
+	/* Load DAC Init Params*/
+	struct no_os_gpio_init_param *gpio_ldac_init;
+	/* Output Range Select */
+	enum ad5754r_dac_ch_range dac_ch_range[AD5754R_NUM_CHANNELS];
+	/* Current Clamp Enable */
+	enum ad5754r_ctrl_current_clamp_en clamp_en;
+	/* Clear Setting */
+	enum ad5754r_ctrl_clear_sel clear_sel;
+	/* SDO Disable */
+	enum ad5754r_ctrl_sdo_dis sdo_dis;
+	/* Powerup States of DAC Channels */
+	enum ad5754r_pwr_dac_ch_pwrup dac_ch_pwr_states[AD5754R_NUM_CHANNELS];
+	/* Powerup State of Internal Reference */
+	enum ad5754r_pwr_int_ref_pwrup int_ref_pwrup;
+	/* Encoding scheme */
+	enum ad5754r_encoding_scheme encoding;
+	/* Reference voltage in millivolts */
+	uint16_t vref;
```

Comment:
if in millivolts, then make it clear in the variable naming -> `vref_mv`

Reply:
Renamed variable